core/src/main/java/org/ironjacamar/core/connectionmanager/pool/AbstractPool.java

```diff
@@ -397,4 +422,49 @@ else if (txCM.getTransactionIntegration().isConnectableResource(mc))
     * @return The instance
     */
    protected abstract ManagedConnectionPool createManagedConnectionPool(Credential credential);
+
+   /**
+    * Prefill the connection pool
+    *
+    * @param credential the pool credential
+    */
+   @Override
+   public void prefill()
+   {
+
+      if (poolConfiguration.isPrefill())
+      {
+         ManagedConnectionPool mcp = pools.get(getPrefillCredential());
+
+         if (mcp == null)
+         {
+            getManagedConnectionPool(getPrefillCredential());
+         } else
+         {
+            mcp.prefill();
```

Comment:
well it's intentional. prefill w/ initial-pool-size is triggered by mcp creation in getManagedConnectionPool(getPrefillCredential())). Explicit call to prefill is needed only on  already present mcp